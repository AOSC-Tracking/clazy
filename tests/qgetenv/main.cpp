#include <QtCore/qglobal.h>
#include <QtCore/qbytearray.h>

int test()
{

    qgetenv("Foo").isEmpty();
    bool b = qgetenv("Foo").isNull();
    QByteArray ba = qgetenv("Foo");
    int fixmeWarnAboutBaseChange = qgetenv("Foo").toInt();
    int fixmeWarnAboutBaseChangeAndKeepOk = qgetenv("Foo").toInt(&b);
    int dontfix = qgetenv("Foo").toInt(&b, 16);
    int fixmeWithoutChangingInternalBase = qgetenv("Foo").toInt(&b, 0);
    int fixmeWithoutChangingInternalBaseAndIgnoringOk = qgetenv("Foo").toInt(nullptr, 0);
    return 0;
}
