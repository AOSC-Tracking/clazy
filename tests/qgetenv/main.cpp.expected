qgetenv/main.cpp:7:5: warning: qgetenv().isEmpty() allocates. Use qEnvironmentVariableIsEmpty() instead [-Wclazy-qgetenv]
qgetenv/main.cpp:8:14: warning: qgetenv().isNull() allocates. Use qEnvironmentVariableIsSet() instead [-Wclazy-qgetenv]
qgetenv/main.cpp:10:36: warning: qgetenv().toInt() is slow. Use qEnvironmentVariableIntValue() instead. This uses internally a base of 0, supporting decimal, hex and octal values [-Wclazy-qgetenv]
qgetenv/main.cpp:11:45: warning: qgetenv().toInt() is slow. Use qEnvironmentVariableIntValue() instead. This uses internally a base of 0, supporting decimal, hex and octal values [-Wclazy-qgetenv]
qgetenv/main.cpp:13:44: warning: qgetenv().toInt() is slow. Use qEnvironmentVariableIntValue() instead [-Wclazy-qgetenv]
qgetenv/main.cpp:14:57: warning: qgetenv().toInt() is slow. Use qEnvironmentVariableIntValue() instead [-Wclazy-qgetenv]
